docs/tutorial/tricks.md

```diff
@@ -0,0 +1,141 @@
+# Common tricks used in (T)FHE
+
+As explained in the [Basics of FHE](../getting-started/fhe_basics.md), the challenge for a developer
+is to adapt his code to fit FHE constraints. In this document we have collected some common examples
+to illustrate the kind of optimization one can do to get better performances.
+
+{% hint style="info" %}
+All code snippets provided here are for educational purpose. In future version of Concrete, some
+functions described here could be directly available in a more generic and efficient form.
+These code snippets are coming from support answers in our [community forum](https://community.zama.ai)
+{% endhint %}
+
+
+## Minimum for Two values
+
+In this first example, we compute a minimum by creating a difference between the two numbers $$y$$ and $$x$$
+and conditionally remove this diff from $$y$$ to either get $$x$$ if $$y>x$$ or $$y$$ if $$x>y$$:
+
+```python
+def min_two(x, y):
+	diff = y - x
+	min_x_y = y - fhe.univariate(lambda x: 0 if x < 0 else x)(diff)
+	return min_x_y
+```
+
+## Maximum for Two values
+
+The companion example of above with the maximum value of two integers instead of the minimum:
+
+```python
+def max_two(x, y):
+	diff = y - x
+	max_x_y = y - fhe.univariate(lambda x: x if x < 0 else 0)(diff)
+	return max_x_y
+```
+
+## Minimum for Three values
+
+And an extension for more than two values:
+
+```python
+def min_three(x, y, z):
+	diff = y - x
+	min_x_y = y - fhe.univariate(lambda x: 0 if x < 0 else x)(diff)
+	diff = z - min_x_y
+	min_x_y_z = z - fhe.univariate(lambda x: 0 if x < 0 else x)(diff)
+
+	return min_x_y_z
+```
+
+## Retrieving a value with an encrypted index
+
+This example show how to deal with an array and an encrypted index. It will create a "selection" array filled with `0` except for the requested index that will be `1`, and sum the products of all array values by this selection array:
+
+```python
+
+ def indexed_value(array, index):
+    all_indices = np.arange(array.size)
+    index_selection = index == all_indices
+    selection_and_zeros = array * index_selection
+    selection = np.sum(selection_and_zeros)
+    return selection
+```
+
+## Filter an array with comparison (>)
+
+This example will filter an encrypted array with a clear condition, here a `greater than`. It will pack all values with a
```

Comment:
We filter it with an encrypted condition actually.

Reply:
fixed